subsys/pm/pm_cpu_shell.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(pm_cpu_shell, CONFIG_PM_LOG_LEVEL);
+
+/* Supported states info from devicetree (CPU0) */
+static const struct pm_state_info residency_info[] =
+	PM_STATE_INFO_LIST_FROM_DT_CPU(DT_NODELABEL(cpu0));
+static size_t residency_info_count = DT_NUM_CPU_POWER_STATES(DT_NODELABEL(cpu0));
+
+/* Enabled cpu idle forced as true, only allow cpu enter soc level pm with 'cpu idle *ms' command */
+static bool pm_cpu_forced_idle = true;
+
+bool pm_cpu_shell_forced_idle(void)
+{
+	return pm_cpu_forced_idle;
+}
+
+static const char *pm_state_to_str(enum pm_state state)
+{
+	switch (state) {
+	case PM_STATE_ACTIVE:
+		return "active";
+	case PM_STATE_RUNTIME_IDLE:
+		return "runtime-idle";
+	case PM_STATE_SUSPEND_TO_IDLE:
+		return "suspend-to-idle";
+	case PM_STATE_STANDBY:
+		return "standby";
+	case PM_STATE_SUSPEND_TO_RAM:
+		return "suspend-to-ram";
+	case PM_STATE_SUSPEND_TO_DISK:
+		return "suspend-to-disk";
+	case PM_STATE_SOFT_OFF:
+		return "soft-off";
+	default:
+		return "UNKNOWN";
+	}
+}
```

Comment:
`pm_state` should be reserved to the pm subsys. Actually this function could be in `pm/state.h`